source/common/quic/envoy_quic_server_session.cc

```diff
@@ -43,7 +43,13 @@ quic::QuicSpdyStream* EnvoyQuicServerSession::CreateIncomingStream(quic::QuicStr
   if (!ShouldCreateIncomingStream(id)) {
     return nullptr;
   }
-  auto stream = new EnvoyQuicServerStream(id, this, quic::BIDIRECTIONAL);
+  if (!codec_stats_.has_value() || !http3_options_.has_value()) {
+    ENVOY_CONN_LOG(error, "Attempt to create stream {} before HCM filter is initialized.", *this,
+                   id);
+    return nullptr;
```

Comment:
Can this actually happen? If it can, can this log spam?

Reply:
OK then please use either ASSERT or ENVOY_BUG depending on what you want to do in terms of providing testing. I don't think kind of error handling makes sense.